docs/pages/database-access/guides/elastic.mdx

```diff
@@ -0,0 +1,248 @@
+
+  See [Installation](../../installation.mdx) for details.
+
+(!docs/pages/includes/user-client-prereqs.mdx!)
+
+## Step 1/5. Install and configure Teleport
+
+### Set up the Teleport Auth and Proxy Services
+
+(!docs/pages/includes/database-access/start-auth-proxy.mdx!)
+
+### Set up the Teleport Database Service
+
+(!docs/pages/includes/database-access/token.mdx!)
+
+Install Teleport on the host where you will run the Teleport Database Service. Make sure you've selected your installation type (OSS, Enterprise, Clo):
+
+(!docs/pages/includes/install-linux.mdx!)
+
+Start the Teleport Database Service, pointing the `--auth-server` flag to the
+address of your Teleport Proxy Service:
+
+<Tabs>
+<TabItem label="Standalone Binary">
+
+```code
+$ teleport db start \
+  --token=/tmp/token \
+  --auth-server=teleport.example.com:3080 \
+  --name=myelastic \
+  --protocol=redis \
+  --uri={ELASTICSEARCH_HOST}:9200 \
+  --labels=env=dev
+```
+
+</TabItem>
+<TabItem label="Config File">
+Add the following block to `teleport.yaml`
+```yaml
+db_service:
+  enabled: true
+  resources:
+  - labels:
+      "*": "*"
+  databases:
+  - name: myelastic
+    protocol: elasticsearch
+    uri: {ELASTICSEARCH_HOST}:9200
+    static_labels:
+      env: dev
+```
+
+Adjust for your environment, then start or restart Teleport.
+
+</TabItem>
+</Tabs>
+
+<Admonition type="note">
+
+The `--auth-server` flag must point to the Teleport cluster's Proxy Service
+endpoint because the Database Service always connects back to the cluster over a
+reverse tunnel.
+
+</Admonition>
+
+<Admonition type="tip">
+  You can start the Database Service using a configuration file instead of CLI flags.
+  See the [YAML reference](../reference/configuration.mdx) for details.
+</Admonition>
+
+## Step 2/5. Create or modify a Teleport user
+
+(!docs/pages/includes/database-access/create-user.mdx!)
+
+Alternately, you can edit an existing user with database access.
+
+1. As a Teleport user with the `editor` role, import a user's configuration to a local `yaml` file:
+
+   ```code
+   $ tctl get user/{teleport-user} > user.yaml
+
+   ```
+
+2. Modify `spec.traits.db_names` and `spec.traits.db_users` with the name(s) and database user(s) this user should have access to, or use `*` to provide access to all databases/users:
+
+    ```yaml
+    ...
+    spec:
+      ...
+      traits:
+      ...
+      db_names:
+      - '*'
+      db_users:
+      - '*'
+      ...
+    ```
+
+3. Apply the changes to the Teleport user:
+
+   ```code
+   $ tctl create -f user.yaml
+   user "teleport-user" has been updated
+   ```
+
+## Step 3/5. Create Role Mapping
+
+Define a role mapping in Elasticsearch to assign your Teleport user or role to an Elasticsearch role.
+The example below maps the Teleport user `alice` to the `superuser` role in Elasticsearch, which
+should **not** be done in production:
```

Comment:
I know it's implied, but I think it would help here to add a note about the kinds of roles a Teleport user should be allowed to have in Elasticsearch.

Reply:
CC @Tener to suggest what roles would be appropriate.

Reply:
This is not a simple thing; the relevant docs are pretty extensive: https://www.elastic.co/guide/en/elasticsearch/reference/current/authorization.html